app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuDrawerFragment.kt

```diff
@@ -17,15 +17,253 @@
  */
 package com.infomaniak.mail.ui.main.menu
 
+import android.content.Intent
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.activity.addCallback
+import androidx.annotation.StringRes
+import androidx.core.view.isGone
+import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.viewModelScope
+import com.infomaniak.lib.core.utils.FormatterFileSize
+import com.infomaniak.lib.core.utils.UtilsUi.openUrl
+import com.infomaniak.lib.core.utils.safeNavigate
+import com.infomaniak.mail.BuildConfig
 import com.infomaniak.mail.R
+import com.infomaniak.mail.data.MailData
+import com.infomaniak.mail.data.models.Folder
+import com.infomaniak.mail.databinding.FragmentMenuDrawerBinding
+import com.infomaniak.mail.ui.LoginActivity
+import com.infomaniak.mail.ui.main.thread.ThreadListFragmentDirections
+import com.infomaniak.mail.ui.main.thread.ThreadListViewModel
+import com.infomaniak.mail.utils.AccountUtils
+import com.infomaniak.mail.utils.toggleChevron
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.launch
+import kotlin.math.ceil
 
-class MenuDrawerFragment : Fragment() {
+class MenuDrawerFragment(
+    private val closeDrawer: (() -> Unit)? = null,
+    private val isDrawerOpen: (() -> Boolean)? = null,
+) : Fragment() {
+
+    private val menuDrawerViewModel: MenuDrawerViewModel by viewModels()
+    private val threadListViewModel: ThreadListViewModel by activityViewModels()
+
+    private lateinit var binding: FragmentMenuDrawerBinding
+
+    private var currentMailboxJob: Job? = null
+    private var mailboxesJob: Job? = null
+    private var foldersJob: Job? = null
```

Comment:
Jobs are supposed to be managed by viewModels, I don't see any exception for their use here.